### ISSUE TITLE
Error in deploying the oauth2.war in the latest product-apim

### ISSUE THREAD

**Issue by DinithHerath** [wso2/api-manager#1317]
### Description

- On the pack startup there is an error log where it states that there is an issue in deploying the oauth2.war in tomcat deployments.

```
ERROR - ContextConfig Missing context.xml: [jar:file:%2FUsers%2Fdinithherath%2FDesktop%2FWork%2Frelease-testing%2F4.2.0%2F405-devportal-admin%2Ffeign-core+issue%2Fwso2am-4.2.0-SNAPSHOT%2Frepository%2Fdeployment%2Fserver%2Fwebapps%2Foauth2.war!/META-INF/context.xml]
java.nio.file.NoSuchFileException: /Users/dinithherath/Desktop/Work/release-testing/4.2.0/405-devportal-admin/feign-core+issue/wso2am-4.2.0-SNAPSHOT/repository/deployment/server/webapps/oauth2.war
    at sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116) ~[?:?]
    at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[?:?]
    at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149) ~[?:?]
    at java.nio.file.Files.readAttributes(Files.java:1764) ~[?:?]
    at java.util.zip.ZipFile$Source.get(ZipFile.java:1273) ~[?:?]
    at java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:845) ~[?:?]
    at java.util.zip.ZipFile$CleanableResource$FinalizableResource.<init>(ZipFile.java:871) ~[?:?]
    at java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:860) ~[?:?]
    at java.util.zip.ZipFile.<init>(ZipFile.java:262) ~[?:?]
    at java.util.zip.ZipFile.<init>(ZipFile.java:191) ~[?:?]
    at java.util.jar.JarFile.<init>(JarFile.java:350) ~[?:?]
    at sun.net.www.protocol.jar.URLJarFile.<init>(URLJarFile.java:103) ~[?:?]
    at sun.net.www.protocol.jar.URLJarFile.getJarFile(URLJarFile.java:72) ~[?:?]
    at sun.net.www.protocol.jar.JarFileFactory.get(JarFileFactory.java:99) ~[?:?]
    at sun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:125) ~[?:?]
    at sun.net.www.protocol.jar.JarURLConnection.getInputStream(JarURLConnection.java:155) ~[?:?]
    at org.apache.catalina.startup.ContextConfig.processContextConfig(ContextConfig.java:716) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.startup.ContextConfig.contextConfig(ContextConfig.java:679) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.startup.ContextConfig.init(ContextConfig.java:949) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:314) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:137) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:173) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:696) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.wso2.carbon.tomcat.internal.CarbonTomcat.addWebApp(CarbonTomcat.java:303) ~[?:?]
    at org.wso2.carbon.tomcat.internal.CarbonTomcat.addWebApp(CarbonTomcat.java:209) ~[?:?]
    at org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer.handleWebappDeployment(TomcatGenericWebappsDeployer.java:255) ~[org.wso2.carbon.webapp.mgt_4.11.6.jar:?]
```
- This blocks the logging process in all portals and first navigate to carbon console.
- Once logged in there, 405 - error pages appears due to the oauth2 not getting deployed.
- Further there are few more error logs as well.
```
[2023-01-24 17:01:29,247]  INFO - RegistryEventingServiceComponent Successfully Initialized Eventing on Registry
[2023-01-24 17:01:29,329]  INFO - JMXServerManager JMX Service URL  : service:jmx:rmi://localhost:11111/jndi/rmi://localhost:9999/jmxrmi
[2023-01-24 17:01:29,884] ERROR - GlobalThrowableMapper An unknown exception has been captured by the global exception mapper.
feign.FeignException$NotFound: [404 ] during [GET] to [https://localhost:9443/oauth2/token/.well-known/openid-configuration] [OpenIDConnectDiscoveryClient#getOpenIdConnectConfiguration()]: []
    at feign.FeignException.clientErrorStatus(FeignException.java:219) ~[io.github.openfeign.feign-core_11.9.1.jar:?]
    at feign.FeignException.errorStatus(FeignException.java:194) ~[io.github.openfeign.feign-core_11.9.1.jar:?]
    at feign.FeignException.errorStatus(FeignException.java:185) ~[io.github.openfeign.feign-core_11.9.1.jar:?]
    at feign.codec.ErrorDecoder$Default.decode(ErrorDecoder.java:92) ~[io.github.openfeign.feign-core_11.9.1.jar:?]
```

### Steps to Reproduce

- Build the latest pack from the master branch.
- Deploy it and try to login into one of the portals.

### Affected Component

APIM

### Version

4.2.0

### Relevant Log Output

```shell
ERROR - ContextConfig Missing context.xml: [jar:file:%2FUsers%2Fdinithherath%2FDesktop%2FWork%2Frelease-testing%2F4.2.0%2F405-devportal-admin%2Ffeign-core+issue%2Fwso2am-4.2.0-SNAPSHOT%2Frepository%2Fdeployment%2Fserver%2Fwebapps%2Foauth2.war!/META-INF/context.xml]
java.nio.file.NoSuchFileException: /Users/dinithherath/Desktop/Work/release-testing/4.2.0/405-devportal-admin/feign-core+issue/wso2am-4.2.0-SNAPSHOT/repository/deployment/server/webapps/oauth2.war
    at sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116) ~[?:?]
    at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[?:?]
    at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149) ~[?:?]
    at java.nio.file.Files.readAttributes(Files.java:1764) ~[?:?]
    at java.util.zip.ZipFile$Source.get(ZipFile.java:1273) ~[?:?]
    at java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:845) ~[?:?]
    at java.util.zip.ZipFile$CleanableResource$FinalizableResource.<init>(ZipFile.java:871) ~[?:?]
    at java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:860) ~[?:?]
    at java.util.zip.ZipFile.<init>(ZipFile.java:262) ~[?:?]
    at java.util.zip.ZipFile.<init>(ZipFile.java:191) ~[?:?]
    at java.util.jar.JarFile.<init>(JarFile.java:350) ~[?:?]
    at sun.net.www.protocol.jar.URLJarFile.<init>(URLJarFile.java:103) ~[?:?]
    at sun.net.www.protocol.jar.URLJarFile.getJarFile(URLJarFile.java:72) ~[?:?]
    at sun.net.www.protocol.jar.JarFileFactory.get(JarFileFactory.java:99) ~[?:?]
    at sun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:125) ~[?:?]
    at sun.net.www.protocol.jar.JarURLConnection.getInputStream(JarURLConnection.java:155) ~[?:?]
    at org.apache.catalina.startup.ContextConfig.processContextConfig(ContextConfig.java:716) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.startup.ContextConfig.contextConfig(ContextConfig.java:679) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.startup.ContextConfig.init(ContextConfig.java:949) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:314) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:137) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:173) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:726) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:698) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:696) ~[tomcat_9.0.58.wso2v1.jar:?]
    at org.wso2.carbon.tomcat.internal.CarbonTomcat.addWebApp(CarbonTomcat.java:303) ~[?:?]
    at org.wso2.carbon.tomcat.internal.CarbonTomcat.addWebApp(CarbonTomcat.java:209) ~[?:?]
    at org.wso2.carbon.webapp.mgt.TomcatGenericWebappsDeployer.handleWebappDeployment(TomcatGenericWebappsDeployer.java:255) ~[org.wso2.carbon.webapp.mgt_4.11.6.jar:?]
```

### Related Issues

https://github.com/wso2/api-manager/issues/1303

**Comment by DinithHerath** [comment]
Last successful commit which triggered successful build is https://github.com/wso2/product-apim/commit/005bb859c4e7d6bf42d8c03d96896b54ed7163e4

**Comment by DinithHerath** [comment]
This is fixed in the latest APIM-4.2.0 build (https://wso2.org/jenkins/job/products/job/product-apim/org.wso2.am$wso2am/3048/). Hence closing the issue now.
